apps/hash-ai-worker-ts/src/activities/flow-activities/write-google-sheet-action/get-filter-from-bp-query-entity.ts

```diff
@@ -0,0 +1,37 @@
+import type { MultiFilter } from "@blockprotocol/graph";
+import { convertBpFilterToGraphFilter } from "@local/hash-backend-utils/convert-bp-filter-to-graph-filter";
+import type { GraphApi } from "@local/hash-graph-client";
+import type { QueryProperties } from "@local/hash-isomorphic-utils/system-types/blockprotocol/query";
+import type { AccountId, Entity, EntityId } from "@local/hash-subgraph";
+
+import { getLatestEntityById } from "../shared/graph-requests";
+
+export const getFilterFromBlockProtocolQueryEntity = async ({
+  authentication,
+  graphApiClient,
+  queryEntityId,
+}: {
+  authentication: { actorId: AccountId };
+  graphApiClient: GraphApi;
+  queryEntityId: EntityId;
+}) => {
+  let queryEntity: Entity<QueryProperties> | undefined;
+  try {
+    queryEntity = (await getLatestEntityById({
+      graphApiClient,
+      authentication,
+      entityId: queryEntityId,
+    })) as Entity<QueryProperties>;
```

Comment:
Nit: maybe we can include a check for whether the entity actually has the correct type, or as part of the query filter by type ID

Reply:
I added a check for whether it has the correct property in https://github.com/hashintel/hash/pull/4357/commits/7e9955151f3f493acd91dca1cd984c6a42ab9e00

I think it's fine to use other entity types as long as they have the query property (although we should replace it with a structural query entity soon)